x/staking/keeper/msg_server.go

```diff
@@ -372,3 +457,574 @@ func (k msgServer) Undelegate(goCtx context.Context, msg *types.MsgUndelegate) (
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delegator := sdk.MustAccAddressFromBech32(msg.DelegatorAddress)
+
+	// If tokenized shares is already disabled, alert the user
+	lockStatus, completionTime := k.GetTokenizeSharesLock(ctx, delegator)
+	if lockStatus == types.TOKENIZE_SHARE_LOCK_STATUS_LOCKED {
+		return nil, types.ErrTokenizeSharesAlreadyDisabledForAccount
+	}
+
+	// If the tokenized shares lock is expiring, remove the pending unlock from the queue
+	if lockStatus == types.TOKENIZE_SHARE_LOCK_STATUS_LOCK_EXPIRING {
+		k.CancelTokenizeShareLockExpiration(ctx, delegator, completionTime)
+	}
+
+	// Create a new tokenization lock for the user
+	// Note: if there is a lock expiration in progress, this will override the expiration
+	k.AddTokenizeSharesLock(ctx, delegator)
+
+	return &types.MsgDisableTokenizeSharesResponse{}, nil
+}
+
+// EnableTokenizeShares begins the countdown after which tokenizing shares by the
+// sender address is re-allowed, which will complete after the unbonding period
+func (k msgServer) EnableTokenizeShares(goCtx context.Context, msg *types.MsgEnableTokenizeShares) (*types.MsgEnableTokenizeSharesResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delegator := sdk.MustAccAddressFromBech32(msg.DelegatorAddress)
+
+	// If tokenized shares aren't current disabled, alert the user
+	lockStatus, unlockTime := k.GetTokenizeSharesLock(ctx, delegator)
+	if lockStatus == types.TOKENIZE_SHARE_LOCK_STATUS_UNLOCKED {
+		return nil, types.ErrTokenizeSharesAlreadyEnabledForAccount
+	}
+	if lockStatus == types.TOKENIZE_SHARE_LOCK_STATUS_LOCK_EXPIRING {
+		return nil, types.ErrTokenizeSharesAlreadyEnabledForAccount.Wrapf(
+			"tokenize shares re-enablement already in progress, ending at %s", unlockTime)
+	}
+
+	// Otherwise queue the unlock
+	completionTime := k.QueueTokenizeSharesAuthorization(ctx, delegator)
+
+	return &types.MsgEnableTokenizeSharesResponse{CompletionTime: completionTime}, nil
+}
+
+// Designates a delegation as a validator bond
+// This enables the validator to receive more liquid staking delegations
+func (k msgServer) ValidatorBond(goCtx context.Context, msg *types.MsgValidatorBond) (*types.MsgValidatorBondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delAddr, err := sdk.AccAddressFromBech32(msg.DelegatorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	valAddr, valErr := sdk.ValAddressFromBech32(msg.ValidatorAddress)
+	if valErr != nil {
+		return nil, valErr
+	}
+
+	validator, found := k.GetValidator(ctx, valAddr)
+	if !found {
+		return nil, types.ErrNoValidatorFound
+	}
+
+	delegation, found := k.GetDelegation(ctx, delAddr, valAddr)
+	if !found {
+		return nil, types.ErrNoDelegation
+	}
+
+	// liquid staking providers should not be able to validator bond
+	if k.DelegatorIsLiquidStaker(delAddr) {
+		return nil, types.ErrValidatorBondNotAllowedFromModuleAccount
+	}
+
+	if !delegation.ValidatorBond {
```

Comment:
This check is unwanted there. It prevents from updating the bond value after another successful delegation.

**Consider this case:**
1) bond factor = 1
2) account delegates 100 ATOM and marks delegation as ValidatorBond
3) now 100 LSM shares will be tokenized
4) to tokenize another 100 shares, the validator bond must be increased (by 100 ATOM respectively)
5) same account delegates another 100 ATOM, his delegation of 200 ATOM is marked as bond, but the actual bond value is still 100 ATOM.
6) **Problem:** there is some confusion created, but what's worse, the bond value cannot be updated by issuing just another `MsgValidatorBond`.

Reply:
Great catch! What do you think is the best path forward here? Two potential options:
* Prevent new delegations or partial undelegations from a validator bond `Delegation` (i.e. prevent any changes to the number of shares of a validator bond `Delegation`)
* Update the validator's `ValidatorBondShares` during delegatoin/undelegation if the underlying delegation is a validator bond

Also curious to hear @jstr1121's opinion

Reply:
> Update the validator's ValidatorBondShares during delegation/undelegation if the underlying Delegation is a validator bond

Second recommendation from Sam would be good to implement I think.

Reply:
I agree with second option too!

If it's hard to implement, simply removing the check is also sufficient - the delegation's bond amount could be updated by an additional `validator-bond` tx, however it might fall far from the original design of that msg :)

Reply:
yeah that's another good option. I think we'd have to tweak it slightly so that it's 
```go
if already validatorBond {
   validator.ValidatorBondShares -= currentDelegatedShares
   validator.ValidatorBondShares += newTotalDelegatedShares
}
```

Would be a simple fix, but would be a bit counterintuitive to know that you have to basically "refresh" the validator bond after a new delegation

Reply:
We actually already update it during undelegation so I think the only change necessary was to update it during delegation.

Added the change [here](https://github.com/iqlusioninc/cosmos-sdk/pull/24)

@xlab @jstr1121